modules/cli/src/main/scala/scala/cli/commands/Run.scala

```diff
@@ -185,10 +185,20 @@ object Run extends ScalaCommand[RunOptions] {
       finally watcher.dispose()
     }
     else {
+      val inputsMd: Inputs = inputs.copy(
+        elements = inputs.elements :+ scala.build.preprocessing.mdsandbox.MdRunner.generateRunnerFile(inputs)
```

Comment:
This should be detected and included in an `Inputs` instance for all commands, instead of being just added here. (with proper handling for allowing  markdown files in place)